Core/Core/Features/Discussions/Editor/ViewModel/DiscussionCreateWebViewModel.swift

```diff
@@ -0,0 +1,111 @@
+    public let urlPathComponent: String = "/discussion_topics/new"
+    public let navigationBarTitle: String
+    public let queryItems: [URLQueryItem]
+
+    private let router: Router
+    private let discussionListViewController: UIViewController?
+
+    /// - parameters:
+    ///   - discussionListViewController: If this variable is present then after creating a discussion the newly created discussion will be pushed using this view controller. If this variable is nil then only the create modal will be dismissed.
+    public init(
+        isAnnouncement: Bool,
+        router: Router = AppEnvironment.shared.router,
+        discussionListViewController: UIViewController?
+    ) {
+        navigationBarTitle = isAnnouncement ? String(localized: "New Announcement", bundle: .core)
+                                            : String(localized: "New Discussion", bundle: .core)
+        queryItems = isAnnouncement ? [URLQueryItem(name: "is_announcement", value: "true")]
+                                    : []
+        self.router = router
+        self.discussionListViewController = discussionListViewController
+    }
+
+    public func webView(
+        _ webView: WKWebView,
+        didStartProvisionalNavigation navigation: WKNavigation!
+    ) {
+        guard let discussionTopicId = webView.url?.discussionTopicId else {
+            return
+        }
+
+        // At this point we are pretty sure that we are navigating to the new discussion
+
+        guard let discussionUrl = webView.url else {
+            return
+        }
+
+        fetchNewDiscussion(
+            discussionUrl: discussionUrl,
+            topicId: discussionTopicId
+        )
+        dismissCreateScreenAndShowNewDiscussion(
+            discussionUrl: discussionUrl,
+            webView: webView
+        )
+    }
+
+    /// Saves the new discussion so it will appear in the discussion list.
+    private func fetchNewDiscussion(
+        discussionUrl: URL,
+        topicId: String
+    ) {
+        guard let context = Context(url: discussionUrl) else {
+            return
+        }
+        let useCase = GetDiscussionTopic(context: context, topicID: topicId)
+        useCase.fetch()
+    }
+
+    private func dismissCreateScreenAndShowNewDiscussion(
+        discussionUrl: URL,
+        webView: WKWebView
+    ) {
+        guard let webViewController = webView.viewController else {
+            return
+        }
+
+        router.dismiss(webViewController) { [router, discussionListViewController] in
+            if let discussionListViewController {
+                router.route(to: discussionUrl, from: discussionListViewController)
+            }
+        }
+    }
+}
+
+extension URL {
+
+    internal var discussionTopicId: String? {
+        guard
+            self.pathComponents.count > 2,
+            self.pathComponents[self.pathComponents.count - 2] == "discussion_topics",
+            let topicIdString = self.pathComponents.last,
```

Comment:
No need to use `self` here. Can we remove those.

```suggestion
            pathComponents.count > 2,
            pathComponents[pathComponents.count - 2] == "discussion_topics",
            let topicIdString = pathComponents.last,
```